src/plugins/workspace/public/plugin.ts

```diff
@@ -3,16 +3,33 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Plugin } from '../../../core/public';
+import type { Subscription } from 'rxjs';
+import { Plugin, CoreStart } from '../../../core/public';
 
 export class WorkspacePlugin implements Plugin<{}, {}, {}> {
+  private coreStart?: CoreStart;
+  private currentWorkspaceSubscription?: Subscription;
+  private _changeSavedObjectCurrentWorkspace() {
+    if (this.coreStart) {
+      return this.coreStart.workspaces.currentWorkspaceId$.subscribe((currentWorkspaceId) => {
+        if (currentWorkspaceId) {
+          this.coreStart?.savedObjects.client.setCurrentWorkspace(currentWorkspaceId);
+        }
+      });
+    }
+  }
   public async setup() {
     return {};
   }
 
-  public start() {
+  public start(core: CoreStart) {
+    this.coreStart = core;
+
+    this.currentWorkspaceSubscription = this._changeSavedObjectCurrentWorkspace();
```

Comment:
how is this currentWorkspaceSubscription used? should it be observable instead?

Reply:
okay, how is this observable currentWorkspaceId$ initialized?

Reply:
The `currentWorkspaceId$` will be initialized by a workspace client. You can find the usage here: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6094